src/random.rs

```diff
@@ -87,7 +86,7 @@ mod derive_serde {
         r.state().1
     }
 
-    #[derive(Clone, Serialize, Deserialize)]
+    #[derive(Copy, Clone, Debug, Serialize, Deserialize)]
```

Comment:
+1 to `Debug` but -1 to `Copy`. The game controls which random generator is used where and records the full history for full replay capabilities. When a generator is copied, I want it to be explicit.